src/lib.rs

```diff
@@ -478,12 +502,14 @@ impl MetadataCommand {
             cmd.current_dir(path);
         }
 
-        if let Some(features) = &self.features {
-            match features {
-                CargoOpt::AllFeatures => cmd.arg("--all-features"),
-                CargoOpt::NoDefaultFeatures => cmd.arg("--no-default-features"),
-                CargoOpt::SomeFeatures(ftrs) => cmd.arg("--features").arg(ftrs.join(",")),
-            };
+        if !self.features.is_empty() {
+            cmd.arg("--features").arg(self.features.join(","));
+        }
+        if self.all_features {
```

Comment:
does `cargo metadata` really accept `--all-features` together with any of the other feature flags?

Reply:
It does. These are equivalent, since `--all-features` supersedes the other feature options:

```
cargo metadata --manifest-path tests/all/Cargo.toml --all-features --no-default-features --features feat2
cargo metadata --manifest-path tests/all/Cargo.toml --all-features
```

Added a test for this usage.